Author: Pino Toscano <pino@debian.org>
Description: Fallback to qmake also when QT_IMPORTS_DIR is "not found".
 If FindQt4.cmake cannot detect the QML imports dir (e.g. because it is
 missing), then QT_IMPORTS_DIR is defined but in a "not found" state.
 Also in such case, fallback to qmake to detect it.
Last-Update: 2012-09-25
Forwarded: no

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,7 +10,7 @@
 
 if( QT_QTDECLARATIVE_FOUND )
   OPTION(IGNORE_CMAKE_INSTALL_PREFIX_FOR_DECLARATIVE_PLUGINS "Install Marble Qt Declarative plugins to QT_IMPORTS_DIR even if that is out of the CMAKE_INSTALL_PREFIX")
-  IF(DEFINED QT_IMPORTS_DIR)
+  IF(DEFINED QT_IMPORTS_DIR AND QT_IMPORTS_DIR)
     SET(MARBLE_QT_IMPORTS_DIR ${QT_IMPORTS_DIR})
   ELSE()
     # Fallback when FindQt4.cmake (part of cmake) does not define QT_IMPORTS_DIR. Get it from qmake instead.
